server/src/main/java/org/elasticsearch/search/fetch/subphase/ExplainPhase.java

```diff
@@ -45,8 +46,17 @@ public void process(HitContext hitContext) throws IOException {
                 for (RescoreContext rescore : context.rescore()) {
                     explanation = rescore.rescorer().explain(topLevelDocId, context.searcher(), rescore, explanation);
                 }
+
                 if (context.rankBuilder() != null) {
-                    explanation = context.rankBuilder().explainHit(explanation, hitContext.rankDoc(), queryNames);
+                    // if we have nested fields, then the query is wrapped using an additional filter on the _primary_term field
+                    // through the DefaultSearchContext#buildFilteredQuery so we have to extract the actual query
+                    if (context.getSearchExecutionContext().nestedLookup() != NestedLookup.EMPTY) {
+                        explanation = explanation.getDetails()[0];
+                    }
+
+                    if (context.rankBuilder() != null) {
+                        explanation = context.rankBuilder().explainHit(explanation, hitContext.rankDoc(), queryNames);
+                    }
```

Comment:
We can remove too since the rank docs query now rely only on the query explain?

Reply:
I think that this is still needed by the existing `rank` for rrf, as well as the semantic-reranker (as we haven't reworked that part yet). Maybe we can cleanup in a subsequent PR ?

Reply:
But this logic was already there since it is used by the `rank` feature? Did we remove it from elsewhere?

Reply:
This was added some time ago to support explain for `rrf`, both for `rank` and `retriever` use case ([related PR](https://github.com/elastic/elasticsearch/pull/108682/files#diff-3990f81b3628cd23afd12570cd14b6c980f4d0221e0c5a2d44bb60290f2d13f1R47)). Since then, it's also used by the semantic reranker as we didn't have access to directly explain at `RankDoc` level.

Reply:
Yep, I was referring to the now removed logic that extract only the first clause of the explanation. All good then.